### ISSUE TITLE
No response in training

### ISSUE THREAD

**Issue by wingsyuan** [marian-nmt/marian#251]
Hi,
When I train the task similar to the example of wmt2017-transformer, I run "sh train.sh --devices 0 1 2 3",but after " Reserving 133 MB, device gpu1", the training can not receive any response.

Can someone give me some advice? Thanks very much!

[2019-02-25 15:02:59] [data] Done shuffling 11941311 sentences to temp files
[2019-02-25 15:03:14] [memory] Reserving 266 MB, device gpu0
[2019-02-25 15:03:14] [memory] Reserving 266 MB, device gpu1
[2019-02-25 15:03:14] [memory] Reserving 266 MB, device gpu3
[2019-02-25 15:03:14] [memory] Reserving 266 MB, device gpu2
[2019-02-25 15:03:15] [memory] Reserving 66 MB, device gpu0
[2019-02-25 15:03:15] [memory] Reserving 66 MB, device gpu1
[2019-02-25 15:03:15] [memory] Reserving 66 MB, device gpu2
[2019-02-25 15:03:15] [memory] Reserving 66 MB, device gpu3
[2019-02-25 15:03:15] [memory] Reserving 266 MB, device gpu1
[2019-02-25 15:03:15] [memory] Reserving 266 MB, device gpu0
[2019-02-25 15:03:15] [memory] Reserving 266 MB, device gpu2
[2019-02-25 15:03:15] [memory] Reserving 266 MB, device gpu3
[2019-02-25 15:03:15] [memory] Reserving 133 MB, device gpu2
[2019-02-25 15:03:15] [memory] Reserving 133 MB, device gpu3
[2019-02-25 15:03:15] [memory] Reserving 133 MB, device gpu0
[2019-02-25 15:03:15] [memory] Reserving 133 MB, device gpu1

script is as follow:
[train-1.txt](https://github.com/marian-nmt/marian/files/2899454/train-1.txt)

**Comment by snukky** [comment]
Try adding `--disp-first 10` and check if you can see display logs for the very first updates.

**Comment by wingsyuan** [comment]
Hi,
@snukky Thanks for your help, the result is as follow:
[2019-02-25 19:27:36] [memory] Reserving 123 MB, device gpu0
[2019-02-25 19:27:36] [memory] Reserving 123 MB, device gpu2
[2019-02-25 19:27:36] [memory] Reserving 123 MB, device gpu3
[2019-02-25 19:27:36] Ep. 1 : Up. 1 : Sen. 1,164 : Cost 10.61644077 : Time 59.15s : 216.47 words/s : L.r. 1.8750e-08
[2019-02-25 19:27:37] Ep. 1 : Up. 2 : Sen. 3,564 : Cost 10.60962009 : Time 0.48s : 44601.38 words/s : L.r. 3.7500e-08

and after that no response again ...
If I use one gpu only ,there is no such problem...

**Comment by wingsyuan** [comment]
When I run "sh train.sh --devices 0", results is as follow
[2019-02-26 10:51:37] [memory] Reserving 238 MB, device gpu0
[2019-02-26 10:51:37] [memory] Reserving 238 MB, device gpu0
[2019-02-26 10:51:37] [memory] Reserving 238 MB, device gpu0
[2019-02-26 10:51:37] [memory] Reserving 477 MB, device gpu0
[2019-02-26 10:51:37] Ep. 1 : Up. 1 : Sen. 512 : Cost 10.54125500 : Time 79.98s : 44.81 words/s : L.r. 1.8750e-08
[2019-02-26 10:51:38] Ep. 1 : Up. 2 : Sen. 760 : Cost 10.51675224 : Time 0.26s : 14151.90 words/s : L.r. 3.7500e-08
[2019-02-26 10:51:38] Ep. 1 : Up. 3 : Sen. 1,008 : Cost 10.53214169 : Time 0.22s : 13560.73 words/s : L.r. 5.6250e-08
[2019-02-26 10:51:38] Ep. 1 : Up. 4 : Sen. 1,256 : Cost 10.53994560 : Time 0.19s : 11966.67 words/s : L.r. 7.5000e-08
[2019-02-26 10:51:38] Ep. 1 : Up. 5 : Sen. 1,504 : Cost 10.53800392 : Time 0.20s : 12342.69 words/s : L.r. 9.3750e-08
[2019-02-26 10:51:39] Ep. 1 : Up. 6 : Sen. 1,752 : Cost 10.53325176 : Time 0.21s : 13903.23 words/s : L.r. 1.1250e-07
[2019-02-26 10:51:39] Ep. 1 : Up. 7 : Sen. 2,264 : Cost 10.53687000 : Time 0.32s : 14193.09 words/s : L.r. 1.3125e-07
[2019-02-26 10:51:39] Ep. 1 : Up. 8 : Sen. 2,512 : Cost 10.53642178 : Time 0.19s : 11882.81 words/s : L.r. 1.5000e-07
[2019-02-26 10:51:39] Ep. 1 : Up. 9 : Sen. 2,760 : Cost 10.52921391 : Time 0.21s : 14073.08 words/s : L.r. 1.6875e-07
[2019-02-26 10:51:40] Ep. 1 : Up. 10 : Sen. 3,272 : Cost 10.55074406 : Time 0.26s : 11718.48 words/s : L.r. 1.8750e-07
[2019-02-26 10:53:50] Ep. 1 : Up. 500 : Sen. 135,801 : Cost 9.73930931 : Time 130.12s : 12336.38 words/s : L.r. 9.3750e-06
[2019-02-26 10:56:07] Ep. 1 : Up. 1000 : Sen. 269,045 : Cost 7.79439163 : Time 137.82s : 12017.87 words/s : L.r. 1.8750e-05
[2019-02-26 10:58:28] Ep. 1 : Up. 1500 : Sen. 407,791 : Cost 6.61105871 : Time 140.22s : 11805.63 

but why no response when I run “sh train.sh --devices 0 1 2 3”:

Is there any ideas? thanks very much!

**Comment by alvations** [comment]
What is the full command you're using? Might be an extra/missing space or missing arguments somewhere.

**Comment by emjotde** [comment]
Hi @wingsyuan 
That message from nvidia-smi clearly says that you have hardware issues, possibly driver problems. My guess, Marian is not causing this, just uncovering.

**Comment by wingsyuan** [comment]
Hi,@emjotde @alvations 

Thanks for your help, 
when I install the newer version of driver(NVIDIA-Linux-x86_64-418.43.run), problems still exists： follow:

[2019-02-26 18:57:12] [data] Shuffling files
[2019-02-26 18:57:36] [data] Done reading 11941311 sentences
[2019-02-26 18:58:15] [data] Done shuffling 11941311 sentences to temp files
[2019-02-26 18:58:30] [memory] Reserving 238 MB, device gpu0
[2019-02-26 18:58:30] [memory] Reserving 238 MB, device gpu1
[2019-02-26 18:58:30] [memory] Reserving 238 MB, device gpu3
[2019-02-26 18:58:30] [memory] Reserving 238 MB, device gpu2
[2019-02-26 18:58:30] [memory] Reserving 59 MB, device gpu0
[2019-02-26 18:58:30] [memory] Reserving 59 MB, device gpu1
[2019-02-26 18:58:30] [memory] Reserving 59 MB, device gpu2
[2019-02-26 18:58:30] [memory] Reserving 59 MB, device gpu3
[2019-02-26 18:58:31] [memory] Reserving 238 MB, device gpu0
[2019-02-26 18:58:31] [memory] Reserving 238 MB, device gpu3
[2019-02-26 18:58:31] [memory] Reserving 238 MB, device gpu2
[2019-02-26 18:58:31] [memory] Reserving 238 MB, device gpu1
[2019-02-26 18:58:31] [memory] Reserving 119 MB, device gpu2
[2019-02-26 18:58:31] [memory] Reserving 119 MB, device gpu0
[2019-02-26 18:58:31] [memory] Reserving 119 MB, device gpu3
[2019-02-26 18:58:31] [memory] Reserving 119 MB, device gpu1
[2019-02-26 18:58:31] Ep. 1 : Up. 1 : Sen. 992 : Cost 10.51303673 : Time 78.93s : 150.82 words/s : L.r. 1.8750e-08
[2019-02-26 18:58:31] Ep. 1 : Up. 2 : Sen. 1,984 : Cost 10.51656342 : Time 0.28s : 35182.53 words/s : L.r. 3.7500e-08
[2019-02-26 18:58:31] Ep. 1 : Up. 3 : Sen. 2,976 : Cost 10.51632023 : Time 0.31s : 44101.97 words/s : L.r. 5.6250e-08

after Up.3 the training can not continue any more...

and when I run "nvidia-smi",the output was still as follow:
nvidia-smi
Unable to determine the device handle for GPU 0000:03:00.0: GPU is lost.  Reboot the system to recover this GPU

Can you help me find the possible cause of this problem? Thanks very much!!!

Here is my command and the script:
chmod u+x train-test.sh
sh train-test.sh 0 1 2 3
[train-test.txt](https://github.com/marian-nmt/marian/files/2905116/train-test.txt)

**Comment by emjotde** [comment]
I don't think we can help you here much. This can be anything from overheating GPU to faulty GPU or bad motherboard firmware. I think we had one such case and needed to replace a faulty GPU. In another case we needed to update the motherboards PCI bridge firmware.

**Comment by wingsyuan** [comment]
Hi,
@emjotde  Thanks for your help.

but I don't think it the hardware problem. As I had try the marian exmple of transformer, the training run well...and  the result is as follow:
[2019-02-27 10:47:48] [memory] Reserving 59 MB, device gpu2
[2019-02-27 10:47:48] [memory] Reserving 59 MB, device gpu3
[2019-02-27 10:47:48] [memory] Reserving 238 MB, device gpu2
[2019-02-27 10:47:48] [memory] Reserving 238 MB, device gpu3
[2019-02-27 10:47:48] [memory] Reserving 238 MB, device gpu1
[2019-02-27 10:47:48] [memory] Reserving 238 MB, device gpu0
[2019-02-27 10:47:48] [memory] Reserving 119 MB, device gpu0
[2019-02-27 10:47:48] [memory] Reserving 119 MB, device gpu3
[2019-02-27 10:47:48] [memory] Reserving 119 MB, device gpu2
[2019-02-27 10:47:48] [memory] Reserving 119 MB, device gpu1
[2019-02-27 10:51:01] Ep. 1 : Up. 500 : Sen. 227,892 : Cost 9.97947311 : Time 214.63s : 32508.81 words/s : L.r. 9.3750e-06
[2019-02-27 10:54:18] Ep. 1 : Up. 1000 : Sen. 460,055 : Cost 8.52352428 : Time 197.23s : 35322.13 words/s : L.r. 1.8750e-05
[2019-02-27 10:57:36] Ep. 1 : Up. 1500 : Sen. 687,899 : Cost 7.89436531 : Time 197.95s : 35227.68 words/s : L.r. 2.8125e-05

at the same time , the output of nvidia-smi is as follow:
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.43       Driver Version: 418.43       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  TITAN Xp            Off  | 00000000:03:00.0 Off |                  N/A |
| 60%   85C    P2   217W / 250W |  10981MiB / 12194MiB |     98%      Default |
+-------------------------------+----------------------+----------------------+
|   1  TITAN Xp            Off  | 00000000:04:00.0 Off |                  N/A |
| 54%   85C    P2   248W / 250W |  10981MiB / 12196MiB |     98%      Default |
+-------------------------------+----------------------+----------------------+
|   2  TITAN Xp            Off  | 00000000:81:00.0 Off |                  N/A |
| 58%   86C    P2   172W / 250W |  10981MiB / 12196MiB |     97%      Default |
+-------------------------------+----------------------+----------------------+
|   3  TITAN Xp            Off  | 00000000:82:00.0 Off |                  N/A |
| 47%   77C    P2   214W / 250W |  10981MiB / 12196MiB |     98%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      8777      C   .../home/xx/marian/build/marian 10971MiB |
|    1      8777      C   .../home/xx/marian/build/marian 10971MiB |
|    2      8777      C   .../home/xx/marian/build/marian 10971MiB |
|    3      8777      C   .../home/xx/marian/build/marian 10971MiB 

it seems that not the hardware fault..

**Comment by emjotde** [comment]
```xxx@DLSY:~$ nvidia-smi
Unable to determine the device handle for GPU 0000:03:00.0: GPU is lost. Reboot the system to recover this GPU1```

This is rather a strong hint. I am quite sure there is no way Marian can cause this.

**Comment by wingsyuan** [comment]
@emjotde  My training script cause this problem， but no problem when I run your example..

**Comment by wingsyuan** [comment]
Hi,@alvations 
After regenerating  the training data, the problem is solved..

when I set the size of vocab.enzh.yml 40000, the output of nvidia-smi is as follow:
nvidia-smi
Unable to determine the device handle for GPU 0000:03:00.0: GPU is lost.  Reboot the system to recover this GPU
the size 36000 is ok..
